Class constructors have obsolete new keyword

Some class constructors have a `new` keyword which is not needed. Only create the `new` keyword for constructors if needed.

Example warning during build: `Warning: /home/runner/work/gir.core/gir.core/src/Libs/Gio-2.0/Public/SocketListener.Constructors.Generated.cs(17,34): warning CS0109: The member 'SocketListener.New()' does not hide an accessible member. The new keyword is not required.`

--- a/src/Generation/Generator/Renderer/Public/Class/ClassConstructors.cs
+++ b/src/Generation/Generator/Renderer/Public/Class/ClassConstructors.cs
@@ -36,7 +36,7 @@ public partial class {cls.Name}
         {
             var parameters = ParameterToNativeExpression.Initialize(constructor.Parameters);
 
-            var newKeyWord = Class.HidesConstructor(cls, constructor)
+            var newKeyWord = Class.HidesConstructor(cls.Parent, constructor)
                 ? "new "
                 : string.Empty;
 

